src/utils/wallet.js

```diff
@@ -182,6 +182,13 @@ export class Wallet {
         }
     }
 
+    async checkContract(contractId) {
```

Comment:
Let's just define this in a Redux action instead of defining in wallet.js and then only calling from Redux action. Generally it applies to almost all stuff in wallet.js except maybe some key-management logic.